File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var check = function(nums) {
    let cnt=0, n= nums.length-1;
    for(let i=1; i<=n; i++){
        if(nums[i-1] > nums[i]) cnt++;
    }
    if(nums[n] > nums[0]) cnt++;
    
    return cnt<=1;
};

File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.cpp
class Solution {
public:
    bool check(vector<int>& nums) {
        int cut=0, len = nums.size()-1;
        
        for(int i=1; i<=len; i++){
            if(nums[i-1] > nums[i]) cut++;
        }
        
        if(nums[len] > nums[0]) cut++;
        
        return cut<=1;
    }
};